crates/kubelet/src/pod/mod.rs

```diff
@@ -211,6 +211,17 @@ impl Pod {
             .unwrap_or_else(|| &EMPTY_VEC)
     }
 
+    /// Get a pod's init containers
+    pub fn init_containers(&self) -> &Vec<KubeContainer> {
+        // We can't use and_then because that requires the lambda to return an Option,
+        // but we have to return an &Option to avoid a move.
+        let inits = match self.0.spec.as_ref() {
+            None => &None,
+            Some(spec) => &spec.init_containers,
+        };
+        inits.as_ref().unwrap_or_else(|| &EMPTY_VEC)
```

Comment:
Hmm, looks like this doesn't work, because `spec.init_containers` is an `Option<Vec<...>>` so doesn't have the `borrow` method.  I think this whole fiddle-faddle was to get around the optionality of `init_containers`...

Reply:
Yeah, I figured it was. Guess we are just doomed to handle the All Optional™ Kubernetes API

Reply:
When I understand Rust better I will probably rage-code something for working with all-optional chains in a less painful way.  `and_then` is meant to support a monadic style but in this case it didn't work either... but if there is a consistent pattern to the k8s APIs then maybe we can figure something out.

"If there is a consistent pattern to the k8s APIs."  Ha ha, what a comedian.

Reply:
Perhaps:

```rust
fn init_containers(&self) -> &[KubeContainer] {
    self.0
        .spec
        .as_ref()
        .and_then(|s| s.init_containers.as_deref())
        .unwrap_or(&[])
}
```

There's not much benefit for returning `&Vec<T>` vs a slice.

Reply:
Ah, `as_deref` might be the secret sauce.  Thanks Peter!

Reply:
It'll return an `Option<&[KubeContainer]>` in this case, though, as `&[T]` is the deref target for `Vec<T>`.

Reply:
If we do want to keep the `&Vec<KubeContainer>` return type, this will work:

```rust
self.0
    .spec
    .as_ref()
    .and_then(|s| s.init_containers.as_ref())
    .unwrap_or(&EMPTY_VEC)
```